emscripten.py

```diff
@@ -1727,9 +1727,11 @@ def create_receiving(function_table_data, function_tables_defs, exported_impleme
     receiving += 'exportAsmFunctions(asm);'
   else:
     if not shared.Settings.SWAPPABLE_ASM_MODULE:
-      if runtime_assertions:
+      if shared.Settings.WASM_ASYNC_COMPILATION and runtime_assertions:
         # assert on the runtime being in a valid state when calling into compiled code. The only
         # exceptions are some support code.
+        # We can't do this trick with synchronous compilation because the `asm` object points to the
+        # modules `exports` which is an immutable object.
```

Comment:
it's only immutable in strict mode or ES6 modules, probably? the comment might be inaccurate.

also, if it is only in those modes, should we not allow the assertions otherwise? (might be risky though, if the user adds a `use strict` manually)

Reply:
No!   It very strange.  The object *always* immutable but without strict mode writes immutable objects silents fail, and have no effect.

Reply:
Oh interesting, so it's not a normal JS object then, or it's a finalized one... I didn't know that.

Another option here might be to make a copy of the object, property by property. That would allow these assertions to still work. However, lgtm as is, maybe just adding a TODO that it's possible to enable the assertions that way.

Reply:
I think its "frozen" maybe?

Reply:
yeah, I'm going to do that option as a followup.